minidump-processor/src/processor.rs

```diff
@@ -356,93 +356,110 @@ where
         .and_then(evil::handle_evil)
         .unwrap_or_default();
 
-    let mut threads = vec![];
-    let mut requesting_thread = None;
-    for (i, thread) in thread_list.threads.iter().enumerate() {
-        let id = thread.raw.thread_id;
-
-        // If this is the thread that wrote the dump, skip processing it.
-        if dump_thread_id.is_some() && dump_thread_id.unwrap() == id {
-            threads.push(CallStack::with_info(id, CallStackInfo::DumpThreadSkipped));
-            continue;
-        }
-
-        let thread_context = thread.context(&dump_system_info, misc_info.as_ref());
-        // If this thread requested the dump then try to use the exception
-        // context if it exists. (prefer the exception stream's thread id over
-        // the breakpad info stream's thread id.)
-        let context = if crashing_thread_id
-            .or(requesting_thread_id)
-            .map(|id| id == thread.raw.thread_id)
-            .unwrap_or(false)
-        {
-            requesting_thread = Some(i);
-            exception_context.as_deref().or(thread_context.as_deref())
-        } else {
-            thread_context.as_deref()
-        };
-
-        let mut stack_memory = thread.stack_memory(&memory_list);
-        // Always chose the memory region that is referenced by the context,
-        // as the `exception_context` may refer to a different memory region than
-        // the `thread_context`, which in turn would fail to stack walk.
-        let stack_ptr = context.as_ref().map(|ctx| ctx.get_stack_pointer());
-        if let Some(stack_ptr) = stack_ptr {
-            let contains_stack_ptr = stack_memory
-                .as_ref()
-                .and_then(|memory| memory.get_memory_at_address::<u64>(stack_ptr))
-                .is_some();
-            if !contains_stack_ptr {
-                stack_memory = memory_list
-                    .memory_at_address(stack_ptr)
-                    .map(Cow::Borrowed)
-                    .or(stack_memory);
-            }
-        }
+    let requesting_thread = std::sync::Mutex::new(None);
+
+    let threads;
+    {
+        let dump_system_info = &dump_system_info;
+        let misc_info = &misc_info;
+        let exception_context = &exception_context;
+        let memory_list = &memory_list;
+        let thread_names = &thread_names;
+        let evil = &evil;
+        let modules = &modules;
+        let system_info = &system_info;
+        let unloaded_modules = &unloaded_modules;
+        let options = &options;
+        let requesting_thread = &requesting_thread;
+        threads = futures_util::future::join_all(thread_list.threads.iter().enumerate().map(
+            |(i, thread)| async move {
```

Comment:
to combat the excessive rightward drift, you can also create a binding for the closure:
`let process_thread = |(i, thread)| async move { … };`
I think you would need some type annotations in that case.